Validation/L1T/plugins/L1Validator.cc

```diff
@@ -82,78 +81,141 @@ void L1Validator::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
-  Handle<L1JetParticleCollection> CenJets;
-  Handle<L1JetParticleCollection> ForJets;
-  Handle<L1JetParticleCollection> Taus;
-  Handle<L1MuonParticleCollection> Muons;
-  //Handle<L1EtMissParticleCollection> METs;
+  Handle<l1t::MuonBxCollection> MuonsBX;
+  Handle<l1t::EGammaBxCollection> EGammasBX;
+  Handle<l1t::TauBxCollection> TausBX;
+  Handle<l1t::JetBxCollection> JetsBX;
+  Handle<GenEventInfoProduct> genEvtInfoProduct;
+  Handle<reco::GenJetCollection> GenJets;
 
   bool GotEverything=true;
 
   if(!iEvent.getByToken(_GenSource, GenParticles)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraIsoEGSource, IsoEGs)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraNonIsoEGSource, NonIsoEGs)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraCenJetSource, CenJets)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraForJetSource, ForJets)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraTauJetSource, Taus)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraMuonSource, Muons)) GotEverything=false;
+  if(!iEvent.getByToken(_L1MuonBXSource, MuonsBX)) GotEverything=false;
+  if(!iEvent.getByToken(_L1EGammaBXSource, EGammasBX)) GotEverything=false;
+  if(!iEvent.getByToken(_L1TauBXSource, TausBX)) GotEverything=false;
+  if(!iEvent.getByToken(_L1JetBXSource, JetsBX)) GotEverything=false;
+  if(!iEvent.getByToken(_srcToken, genEvtInfoProduct)) GotEverything=false;
+  if(!iEvent.getByToken(_L1GenJetSource, GenJets)) GotEverything=false;  
 
   if(!GotEverything) return;
 
+  /*
+  std::string moduleName = "";
+  if( genEvtInfoProduct.isValid() ) {
+	  const edm::Provenance& prov = iEvent.getProvenance(genEvtInfoProduct.id());
+	  moduleName = edm::moduleName(prov);
+	  //cout<<" generator name: "<<moduleName<<endl;
+  }
+  */ 
 
   _Hists.NEvents++;
 
-  _Hists.FillNumber(L1ValidatorHists::Type::IsoEG, IsoEGs->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::NonIsoEG, NonIsoEGs->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::CenJet, CenJets->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::ForJet, ForJets->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::TauJet, Taus->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::Muon, Muons->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Muon, MuonsBX->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Egamma, EGammasBX->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Tau, TausBX->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Jet, JetsBX->size());
+
+  //For gen jet
+
+  for(uint i=0; i < GenJets->size(); i++){
+     const reco::GenJet *Genjet = &GenJets->at(i);
+     if(fabs(Genjet->eta())>4.7) continue;
+
+     double minDR = 999.0;
+     const l1t::Jet        *L1Part=NULL;
+     for(int iBx = JetsBX->getFirstBX();  iBx<=JetsBX->getLastBX(); ++iBx){
+          if(iBx>0) continue;
+          for(std::vector<l1t::Jet>::const_iterator jet = JetsBX->begin(iBx); jet != JetsBX->end(iBx); ++jet){
```

Comment:
@davidlange6  In this case, could we stay with the old style for loop, unless strong arguments against it ? 

I would agree that there are readability arguments for range-based for loops.
However, I have not seen it used with BXVector. ( JetBx is a BXVector pointer.)
BXVector has begin( ), end( ), begin(int), end(int) methods, but its use in range-based loops has not been tested, or I don't remember one.